pkg/k8s/watchers/pod.go

```diff
@@ -72,6 +74,17 @@ func (k *K8sWatcher) createPodController(getter cache.Getter, fieldSelector fiel
 				var valid bool
 				if pod := k8s.ObjTov1Pod(obj); pod != nil {
 					valid = true
+					podNSName := k8sUtils.GetObjNamespaceName(&pod.ObjectMeta)
+					ep := k.endpointManager.LookupPodName(podNSName)
+					// If ep is not nil then we have received the CNI event
+					// first and the k8s event afterwards, if this happens it's
+					// likely the Kube API Server is getting behind the event
+					// handling.
+					if ep != nil {
+						now := time.Now()
+						epCreatedAt := ep.GetCreatedAt()
+						metrics.EventLagK8s.Set(now.Sub(epCreatedAt).Seconds())
+					}
```

Comment:
```suggestion
					if ep != nil {
						epCreatedAt := ep.GetCreatedAt()
						metrics.EventLagK8s.Set(time.Since(epCreatedAt).Seconds())
					}
```

https://golang.org/pkg/time/#Since